### PR TITLE
Add a docker environment

### PR DESCRIPTION
**Pull request by Crease29**
### 1. Why is this change necessary?

Add a docker environment for development.

### 2. What does this change do, exactly?

It adds a docker environment for development.

### 5. Checklist

- [X] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

File: htdocs/info.php [hunk 1]
```diff
@@ -0,0 +1 @@
+<?php phpinfo();
```
Comment (teiling88) [inline htdocs/info.php]: bitte raus nehmen - sonst landet das 1zu1 aufm Live Server ;-)

File: .psh.yaml.dist [hunk 2]
```diff
@@ -32,11 +32,27 @@ const:
...
 
 environments:
+  docker:
+    paths:
+      - "dev-ops/docker/actions"
+    const:
+      DB_HOST: "mariadb"
+      FRONTEND_URL: "http://docker.team-opencaching.de"
+      SESSION_DOMAIN: "docker.team-opencaching.de"
+    templates:
```
Comment (teiling88) [inline .psh.yaml.dist:42]: bitte schauen ob nicht die Templates aus dem local.team-opencaching.de verwendet werden können

File: dev-ops/docker/actions/init.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+#DESCRIPTION: initialisation process to start developing
+
+INCLUDE: ./composer.sh
+INCLUDE: ./.init-database.sh
+INCLUDE: ./import-stored-proc.sh
+INCLUDE: ./import-sql-static.sh
+INCLUDE: ./sf-migrations.sh
+INCLUDE: ./okapi-update.sh
+INCLUDE: ./import-translations.sh
```
Comment (teiling88) [inline dev-ops/docker/actions/init.sh]: Bitte prüfen ob Du die Sachen nicht aus local.team-opencaching.de Ordner includen kannst :-)

**Review by teiling88 — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        networks:
+            - opencaching
+
+    webserver:
+        image: httpd:2.4.37-alpine
+        container_name: opencaching-webserver
+        restart: on-failure
+        working_dir: /application
+        links:
+            - php-fpm
```
Comment (nlubisch) [inline docker-compose.yml:31]: Links are a legacy feature and may be removed in upcoming Docker releases.
See: https://docs.docker.com/compose/compose-file/#links

Comment (Crease29) [reply docker-compose.yml:31]: Fixed in a6874825a6522f231a86d6fac7860f41c70cc51f

**Pull request merged**